swift/apple/Firezone/Application/FirezoneApp.swift

```diff
@@ -86,6 +91,49 @@ struct FirezoneApp: App {
       maybeShowOutdatedAlert()
     }
 
+    private func enforceSingleInstance() {
+      // Get the actual bundle identifier from the running app
+      guard let bundleId = Bundle.main.bundleIdentifier else { return }
+
+      let runningApps = NSRunningApplication.runningApplications(
+        withBundleIdentifier: bundleId
+      )
+
+      guard runningApps.count > 1 else { return }
+
+      for app in runningApps where app != NSRunningApplication.current {
+        // Try to terminate the other instance
+        var terminated = app.terminate()
+
+        if !terminated {
+          // If graceful termination fails, force terminate
+          terminated = app.forceTerminate()
+        }
+
+        if !terminated {
+          // If we still can't terminate, show an alert to the user
+          DispatchQueue.main.async {
+            let alert = NSAlert()
+            alert.messageText = "Another Firezone Instance Detected"
+            alert.informativeText = """
+              Another instance of Firezone is already running. \
+              Please quit the other instance manually from the menu bar to continue.
```

Comment:
Minor suggestion on wording simplification.

```suggestion
              Please quit the other instance from the menu bar to continue.
```

Reply:
Another option could be

> Firezone is already running. Quit the other one from the menu bar to continue.

Just because these users are usually non technical and many times not native English speakers.